app/routes/goal_routes.py

```diff
@@ -1 +1,114 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import abort, Blueprint, make_response, request, Response
+from app.models.task import Task
+from app.models.goal import Goal
+from ..db import db
+from sqlalchemy import cast, String, Integer
+from .route_utilities import validate_model
+from datetime import datetime
+import os
+import requests
+
+bp = Blueprint("goals_bp", __name__, url_prefix = "/goals")
+
+
+@bp.post("") 
+def create_goal():  
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal.from_dict(request_body)
+
+    except KeyError as error:
+        response = {"details": "Invalid data"}
+        abort(make_response(response, 400))
+        
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_dict()}, 201
+
+@bp.get("/<goal_id>")
+def get_one_goal_by_id(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    # response = [goal.to_dict() for goal in goal.goals]
+    return {"goal": goal.to_dict()}, 200
+
+@bp.get("")
+def get_all_goals():
+    query = db.select(Goal)
+    goals = db.session.execute(query).scalars().all()
```

Comment:
@ra-choa, I am curious as to what this line's purpose is? The pattern isn't what we've showed in the curriculum to execute a query. Also, it doesn't seem like you are using the variable after its initialization (you just build another query and then assign its result to `goals` on `line 53`. Is this just a relic of a preceding implementation? **Please respond to provide some clarity**.